modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDynamicParameterTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.engine;
+
+import java.math.BigDecimal;
+import java.sql.Date;
+import java.time.LocalDate;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+/** Dynamic parameters checks. */
+public class ItDynamicParameterTest extends AbstractBasicIntegrationTest {
```

Comment:
Let's extend test coverage with the following scenarios: 
1) for every supported type, query `SELECT ?` returns proper metadata type
2) usage of dynamic params in FROM, WHERE, LIMIT and OFFSET clauses

Besides, the query `assertQuery("SELECT typeof(?) = typeof(2)").withParams((short)2).returns(true).check();` fails

Reply:
thanks. Fixed.